web-client/instructions/Preparing the native builds.md

```diff
@@ -0,0 +1,85 @@
+# Preparing the native builds
+
```

Comment:
I ran into trouble trying to get `npm exec ionic` working: apparently that ends up installing an older, deprecated Ionic CLI package, which fails to work without much clear explanation.

How does this read as a note to avoid that?

````suggestion

## CLI note

To run these commands without installing the Ionic CLI globally, run:

```
npm exec @ionic/cli capacitor […]
```

Running `npm exec ionic capacitor […]` will **not** work:
this will install and run a deprecated version of the CLI.

See: [Install the Ionic CLI](https://ionicframework.com/docs/intro/cli#install-the-ionic-cli).

````

Reply:
actually `npx capacitor [option]` can be run instead. There is no need to use the ionic CLI

Reply:
Hmm, that doesn't seem to work here:

```
$ npm exec capacitor build
[error] Unknown command: build

$ npx capacitor --version
3.4.0
```

(Or `build` isn't a supported command, when run this way?)

Reply:
`npx capacitor build`

Reply:
That still behaves the same: I was actually running `npx` locally, sorry. (`npx` should just `npm exec`, under the hood, these days.)

Reply:
```
➜  web-client git:(release-instructions) npx capacitor      

  ⚡️  Capacitor - Cross-Platform apps with JavaScript and the Web  ⚡️

Usage: capacitor [options] [command] [command]

Options:
  -V, --version                     output the version number
  -h, --help                        display help for command

Commands:
  init [options] [appName] [appId]  Initialize Capacitor configuration
  sync [options] [platform]         copy + update
  update [options] [platform]       updates the native plugins and dependencies based on package.json
  copy [platform]                   copies the web app build into the native app
  run [options] [platform]          runs sync, then builds and deploys the native app
  open [platform]                   opens the native project workspace (Xcode for iOS)
  add [platform]                    add a native platform project
  ls [platform]                     list installed Cordova and Capacitor plugins
  doctor [platform]                 checks the current setup for common errors
```

Reply:
Yeah, the `capacitor` command seems to differ from the `ionic capacitor` subcommand: `capacitor` seems to have no `build` subcommand. (It's not listed in <https://capacitorjs.com/docs/cli> either.)

Reply:
To build, the `angular cli` can be used because the `ionic cli` uses it anyway

Reply:
> Yeah, the `capacitor` command seems to differ from the `ionic capacitor` subcommand: `capacitor` seems to have no `build` subcommand. (It's not listed in https://capacitorjs.com/docs/cli either.)

That's because we build the `web` app with ionic/angular CLI, then we copy the web into the `native` platforms with `Capacitor`.

Think of Ionic as tailwind or bootstrap

Reply:
The `ionic capacitor` commands invoke the `framework command` + `capacitor [option]`

Reply:
```
➜  web-client git:(release-instructions) ionic capacitor

  ionic capacitor - Capacitor functionality

    These commands integrate with Capacitor, Ionic's new native layer project which provides an alternative to Cordova
    for native functionality in your app.
    
    Learn more about Capacitor:
    - Main documentation: https://ion.link/capacitor

  Usage:

    $ ionic capacitor <command> [<args>] [--help] [--verbose] [--quiet] [--no-interactive] [--no-color] [--confirm] [options]

  Project Commands:

    add ............................. Add a native platform to your Ionic project
    build ........................... (beta) Build an Ionic project for a given platform
    copy ............................ Copy web assets to native platforms
    open ............................ Open the IDE for a given native platform project
    run ............................. Run an Ionic project on a connected device
    sync ............................ Sync (copy + update) an Ionic project
    update .......................... Update Capacitor native platforms, install Capacitor/Cordova plugins
```